### PR TITLE
Convert P6 tests to also run for P7 and onwards

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Prior to this PR, numerous of our tests are only targeting protocol version 6, this PR changes this such that these tests are generalized to run against every protocol version starting from P6 and onwards.

## Changes

- No changes are intended by this PR besides generalizing the tests, the only exception is a test related to account nonce https://github.com/Concordium/concordium-node/pull/1111/files#diff-176f6e3e32412c792f8e0873ce966e99076d1be8da6a228d9226d6b87301fbefR760

## Checklist

- [ ] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Review by abizjak — APPROVED** [review]
It's a bit annoying to review this since a lot of the changes are just whitespace changes.

But looks good.

File: concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/Common.hs [hunk 1]
```diff
@@ -77,3 +135,34 @@ sigKeyPair = fst $ Dummy.randomBlockKeyPair $ mkStdGen 42
...
 sigPublicKey = Sig.verifyKey sigKeyPair
+
+-- | Run tests for each protocol version.
+-- The tests are provided with the protocol version and a string representation of the protocol version.
+forEveryProtocolVersion ::
+    (forall pv. (IsProtocolVersion pv) => SProtocolVersion pv -> String -> Spec) ->
+    Spec
+forEveryProtocolVersion check =
+    sequence_
+        [ check SP1 "P1",
```
Comment (EmilLa1) [inline concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/Common.hs:146]: Why don't we just use the `Show` instance for the `ProtocolVersion` ?

**Review by EmilLa1 — APPROVED** [review]
Looks good, great job! :+1:

File: concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/Common.hs (hunk 1 above)
Comment (limemloh) [reply concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/Common.hs:146]: If found this simpler, it becomes something like this then:
```suggestion
        [ check SP1 (show $ demoteProtocolVersion SP1),
```
I don't mind changing it, if you prefer this.

**Pull request merged**